kuzzleio/documentation
# KZL-1124-add-selector-on-api-page

## What does this PR do?

[KZL-1124](https://jira.kaliop.net/browse/KZL-1124)

**API PAGE:**
When browsing API ~~methods~~, add a sdk selector displaying ~~only SDK wich contains the method (selector not displayed if no one contains the method).~~ all SDK. 
If browsing API methods, sdk which not contains method aren't available. It redirect on sdk method page. 
If browsing anything else, all sdk are available but redirect on sdk `getting-started` page.

**SDK PAGE:**
When browsing SDK methods, add a 'See API doc' button in SDK selector.
If browsing SDK methods, redirect to API method page.
If browsing anything else, redirect to `connecting to kuzzle` page.

~~Not displayed on mobile view.~~ 
Both available on mobile view in sidebar (see screen below).


Following screen were made without submodules init (that's why SDK JS6 isn't available).
**Computer**

_API_

Browsing `Connecting to Kuzzle` page. Click on an sdk redirect on sdk`getting-started` page.
![image](https://user-images.githubusercontent.com/44427849/61383560-37c65a00-a8af-11e9-8d06-6dd3c1ccbba0.png)

Browsing `admin/dump` page. Can't click on any sdk because no one contains admin/dump method.
![image](https://user-images.githubusercontent.com/44427849/61383703-7825d800-a8af-11e9-9d80-921616652681.png)

Browsing `auth/login` page. Click on a sdk redirect on auth/login sdk method page.
![image](https://user-images.githubusercontent.com/44427849/61383780-9f7ca500-a8af-11e9-82a5-8f1efedcf173.png)

_SDK_

Browsing cpp sdk `getting-started` page. Click on API button redirect on `Connecting to Kuzzle` api page. Click on any sdk redirect on `getting-started` sdk page.
![image](https://user-images.githubusercontent.com/44427849/61383929-e36faa00-a8af-11e9-8cd3-eb4aad1accab.png)

Browsing cpp sdk `auth/login` page. Click on Api button redirect on same method but api page. Same for SDK.
![image](https://user-images.githubusercontent.com/44427849/61384077-27fb4580-a8b0-11e9-9305-bb550627d624.png)

**Mobile**
Browsing on mobile view -> same as before
![image](https://user-images.githubusercontent.com/44427849/61384162-58db7a80-a8b0-11e9-8ac2-89dc384920b2.png)


### How should this be manually tested?

  - Step 1 : npm run dev 
  - Step 2 : (computer view) go to API auth login for example
  - Step 3 : use Selector on the right to check the SDK page of the auth login method

### Other changes
Add little transition when opening selector.

### Boyscout
/
